src/main/groovy/org/transmartproject/batch/support/ScientificNotationFormat.groovy

```diff
@@ -20,7 +20,8 @@ class ScientificNotationFormat extends NumberFormat {
         Number result = decimalFormat.parse(normalizedString, pos)
 
         if (pos.index < normalizedString.length()) {
-            throw new IllegalArgumentException("Number (${text}) contains unrecognized characters.")
+            throw new IllegalArgumentException(
+                    "${normalizedString} contains unrecognized character at ${pos.index + 1} position to be a number.")
```

Comment:
This sentence doesn't seem gramatically correct. Maybe "${normalizedString} cannot be parsed as a number owing to an unrecognized character at position `${pos.index + 1}"`